CRM/Logging/ReportDetail.php

```diff
@@ -228,10 +231,30 @@ protected function convertDiffsToRows() {
           $to = '';
         }
       }
-
-      $rows[] = ['field' => $field . " (id: {$diff['id']})", 'from' => $from, 'to' => $to];
+      // We don't need that field displayed
+      if ($field == 'Modified Date') {
+        $nRows[$diff['id']][] = ['field' => '', 'from' => $from, 'to' => $to];
+      }
+      else {
+        $nRows[$diff['id']][] = ['field' => $field . " (id: {$diff['id']})", 'from' => $from, 'to' => $to];
+      }
     }
+    // Transform the output so that we can compact the changes into the proper
+    // amount of rows IF trData is holding more than 1 array
+    foreach ($nRows as $trData) {
+      if (count($trData) > 1) {
+        $keys = array_intersect(...array_map('array_keys', $trData));
+        $mergedRes = array_combine($keys, array_map(function ($key) use ($trData) {
+          // Lets use a line break
+          return implode('<br/>', array_column($trData, $key));
```

Comment:
It's not a bad idea and looks nice on screen, but I wonder if this is going to thwart sanitizing or other processing/hooks. I haven't tried this but something like update CRM/Report/Form/Layout.tpl to know what to do with subarrays for `$row.field`, then could assign an array as the value here, and let the tpl format the subarray with `<br>` (or maybe `<div>`).

Reply:
Well, my approach was to go as minimal/quiet as i could in order to maintain the same exact functionality without harassing tpls etc. I am just afraid that if i start poking the CRM/Report/Form/Layout/Table.tpl, things might blow :thinking: 

You are talking about taking care of the array grouping thingie that I introduced somewhere around here:
https://lab.civicrm.org/dev/core/-/blob/master/templates/CRM/Report/Form/Layout/Table.tpl#L94 ?

Reply:
Yeah inside that loop somewhere, maybe just above the if Subtotal. Something like if is_array, then loop and surround by divs. Might be trickier to do if the subarray isn't strings. I admit it might take some fiddling to get right.

Reply:
Following your suggestion, I managed to make it behave exactly the same but by doing the concatenation in the tpl instead: by sending the subarrays to the tpl and then intercepting the case that the source is an array!
Way more clearer now, they do render nicely as before, enclosed in divs

![image](https://user-images.githubusercontent.com/14973113/97990770-561fd400-1dd8-11eb-81c4-5bc955e678a3.png)

I had to do exactly the same alter in the `overlay.tpl` as well, to cover the case of hover/mouseover scenario.

I do think that right now we still do keep the compatibility as it was while at the same time enhancing it.

What do you think ?